File: solutions/clojure/sublist/1/src/sublist.clj
(ns sublist)

(defn classify [list1 list2] ;; <- arglist goes here
  (let [sublist (fn [lst1 lst2]
                  (some #(= % lst1)
                        (partition (count lst1) 1 lst2)))]
    (cond
      (= list1 list2) :equal
      (sublist list1 list2) :sublist
      (sublist list2 list1) :superlist
      :else                 :unequal)))

File: solutions/elixir/sublist/1/lib/sublist.ex
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """
  def compare(a, b) do
    len_a = Enum.count(a)
    len_b = Enum.count(b)

    cond do
      a === b ->
        :equal

      a === [] ->
        :sublist

      b === [] ->
        :superlist

      len_a > len_b ->
        a
        |> Enum.chunk_every(len_b, 1)
        |> Enum.find_value(:unequal, fn chunk -> if chunk === b, do: :superlist end)

      len_b > len_a ->
        b
        |> Enum.chunk_every(len_a, 1)
        |> Enum.find_value(:unequal, fn chunk -> if chunk === a, do: :sublist end)

      true ->
        :unequal
    end
  end
end
